lib/api/apiUtils/bucket/bucketCors.js

```diff
@@ -0,0 +1,278 @@
+import { parseString } from 'xml2js';
+
+import { errors } from 'arsenal';
+
+/*
+    Format of xml request:
+
+    <CORSConfiguration>
+        <CORSRule>
+            <AllowedOrigin>http://www.example.com</AllowedOrigin>
+            <AllowedMethod>PUT</AllowedMethod>
+            <AllowedMethod>POST</AllowedMethod>
+            <AllowedMethod>DELETE</AllowedMethod>
+            <AllowedHeader>*</AllowedHeader>
+            <MaxAgeSeconds>3000</MaxAgeSec>
+            <ExposeHeader>x-amz-server-side-encryption</ExposeHeader>
+        </CORSRule>
+        <CORSRule>
+            <AllowedOrigin>*</AllowedOrigin>
+            <AllowedMethod>GET</AllowedMethod>
+            <AllowedHeader>*</AllowedHeader>
+            <MaxAgeSeconds>3000</MaxAgeSeconds>
+        </CORSRule>
+    </CORSConfiguration>
+*/
+
+const errs = {
+    numberRules: 'The number of CORS rules should not exceed allowed limit ' +
+    'of 100 rules.',
+    originAndMethodExist: 'Each CORSRule must identify at least one origin ' +
+    'and one method.',
+};
+
+const _utility = {
+    validateID(id) {
+        const errMsg = 'The ID value can be up to 255 characters long. ' +
+        `Invalid ID is "${id}"`;
+        if (!id) {
+            return undefined;
+        }
+        if (!Array.isArray(id) || id.length !== 1
+        || typeof id[0] !== 'string') {
+            return errors.MalformedXML;
+        }
+        if (id[0] === '') {
+            return undefined;
+        }
+        // NOTE: AWS does not make this check, but the spec specifies IDs should
+        // be up to 255 characters in length.
+        // http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTcors.html
+        if (id[0].length > 255) {
+            return errors.InvalidRequest.customizeDescription(errMsg);
+        }
+        return true; // to indicate ID exists
+    },
+    validateMaxAgeSeconds(seconds) {
+        if (!seconds) {
+            return undefined;
+        }
+        if (!Array.isArray(seconds) || seconds.length !== 1) {
+            return errors.MalformedXML;
+        }
+        if (seconds[0] === '') {
+            return undefined;
+        }
+        const parsedValue = parseInt(seconds[0], 10);
+        const errMsg = `MaxAgeSeconds "${seconds[0]}" is not a valid value.`;
+        if (isNaN(parsedValue) || parsedValue < 0) {
+            return errors.MalformedXML.customizeDescription(errMsg);
+        }
+        return parsedValue;
```

Comment:
I tried to make them a little more consistent now so that all return 
- true if valid and validated argument exists, 
- undefined if argument does not exist but valid because optional, and 
- Error if invalid. 

The one exception is maxAgeSeconds because I parse the value in the validation function and figure I might as well return it instead of reparsing the int in the caller function.